### ISSUE TITLE
Ratios with float endpoints might have wrong definition

### ISSUE THREAD

**Issue by mrjbq7** [factor/factor#2880]
```factor
{ { 1.5 2.5 3.5 } } [ 1.5 3.7 [a..b) >array ] unit-test
```

**Comment by mrjbq7** [comment]
More test cases:

```factor
{ { 1+1/2 2+1/2 3+1/2 } } [ 3/2 37/10 [a..b) >array ] unit-test
```

And maybe a patch that makes these work:

```diff
diff --git a/core/ranges/ranges.factor b/core/ranges/ranges.factor
index 007444abfc..da7176a90f 100644
--- a/core/ranges/ranges.factor
+++ b/core/ranges/ranges.factor
@@ -49,7 +49,8 @@ M: range supremum
 
 : (a.. ( a b step -- a' b' step ) dup [ + ] curry 2dip ; inline
 
-: ..b) ( a b step -- a' b' step ) dup [ - ] curry dip ; inline
+: ..b) ( a b step -- a' b' step )
+    2over - over mod zero? [ dup [ - ] curry dip ] when ; inline
 
 PRIVATE>
 
```

**Comment by mrjbq7** [comment]
Fixed in 536a4e951d56d4b2307fbd477952648e202b9ba3.
